fortnite_api/cosmetics/lego.py

```diff
@@ -0,0 +1,120 @@
+"""
+MIT License
+
+Copyright (c) 2019-present Luc1412
+
+Permission is hereby granted, free of charge, to any person obtaining a copy
+of this software and associated documentation files (the "Software"), to deal
+in the Software without restriction, including without limitation the rights
+to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+copies of the Software, and to permit persons to whom the Software is
+furnished to do so, subject to the following conditions:
+
+The above copyright notice and this permission notice shall be included in all
+copies or substantial portions of the Software.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+SOFTWARE.
+"""
+
+from __future__ import annotations
+
+from typing import TYPE_CHECKING, Any, Coroutine, Dict, List, Optional, Tuple, Union, overload
+
+from ..asset import Asset
+from ..enums import GameLanguage
+from ..http import HTTPClientT
+from ..utils import get_with_fallback
+from .br import CosmeticBr
+from .common import Cosmetic, CosmeticImages
+
+if TYPE_CHECKING:
+    from ..http import HTTPClient, SyncHTTPClient
+
+__all__: Tuple[str, ...] = ('CosmeticLego', 'CosmeticLegoImages')
+
+
+class CosmeticLegoImages(CosmeticImages[HTTPClientT]):
+    """
+    .. attributetable:: fortnite_api.CosmeticLegoImages
+
+    A class representing the images of a lego cosmetic.
+
+    This class inherits from :class:`fortnite_api.CosmeticImages`.
+
+    Attributes
+    ----------
+    wide: Optional[:class:`fortnite_api.Asset`]
+        The wide image of the lego cosmetic.
+    """
+
+    def __init__(self, *, data: Dict[str, Any], http: HTTPClientT) -> None:
+        super().__init__(data=data, http=http)
+
+        _wide = data.get('wide')
+        self.wide: Optional[Asset[HTTPClientT]] = _wide and Asset(http=http, url=_wide)
+
+
+class CosmeticLego(Cosmetic[HTTPClientT]):
```

Comment:
Not sure about the naming, since it's not a real cosmetic, rather a Lego Version of an existing cosmetic.

Reply:
Regardless of where the cosmetic stems from it is nonetheless a cosmetic because it's a usable item in Fortnite Lego, no?

Reply:
I mean you buy a specific skin in the Shop. If you play normal modes, you play the normal variant. If you play LEGO modes, the lego variant. This CosmeticLego, describes this LEGO variant.

Reply:
Just `Lego` then perhaps? This would require the class to be moved out of `/cosmetics` and all the docs to be updated too if the API does not consider it a cosmetic

Reply:
I think Lego isn't a good name. Rather `LegoVariant`.